src/Extension/IntlExtension.php

```diff
@@ -0,0 +1,393 @@
+        }
+
+        return Locales::getName($data, $locale);
+    }
+
+    public function getTimezoneName(string $timezone, string $locale = null): string
+    {
+        if (!class_exists(Timezones::class)) {
+            throw new \LogicException('The "timezone_name" filter requires the symfony/intl package to be installed.');
+        }
+
+        return Timezones::getName($timezone, $locale);
+    }
+
+    public function getCountryTimezones(string $country): array
+    {
+        if (!class_exists(Timezones::class)) {
+            throw new \LogicException('The "country_timezone" filter requires the symfony/intl package to be installed.');
+        }
+
+        return Timezones::forCountryCode($country);
+    }
+
+    public function formatCurrency($amount, string $currency, string $locale = null, array $attrs = []): string
+    {
+        if (!class_exists(\Locale::class)) {
+            throw new RuntimeException('The "format_currency" filter requires the PHP intl extension or the symfony/polyfill-intl-icu package to be installed.');
+        }
+
+        $formatter = $this->createNumberFormatter($locale, 'currency', $attrs);
+
+        if (false === $ret = $formatter->formatCurrency($amount, $currency)) {
+            throw new RuntimeError('Unable to format the given number as a currency.');
+        }
+
+        return $ret;
+    }
+
+    public function formatNumber($number, string $locale = null, array $attrs = [], string $style = 'decimal', string $type = 'default'): string
+    {
+        if (!class_exists(\Locale::class)) {
+            throw new RuntimeException('The "format_number" filter requires the PHP intl extension or the symfony/polyfill-intl-icu package to be installed.');
+        }
+
+        if (!isset(self::NUMBER_TYPES[$type])) {
+            throw new RuntimeError(sprintf('The type "%s" does not exist, known types are: "%s".', $type, implode('", "', array_keys(self::NUMBER_TYPES))));
+        }
+
+        $formatter = $this->createNumberFormatter($locale, $style, $attrs);
+
+        if (false === $ret = $formatter->format($number, self::NUMBER_TYPES[$type])) {
+            throw new RuntimeError('Unable to format the given number.');
+        }
+
+        return $ret;
+    }
+
+    public function formatNumberStyle(string $style, $number, string $locale = null, array $attrs = [], string $type = 'default'): string
+    {
+        return $this->formatNumber($number, $locale, $attrs, $style, $type);
+    }
+
+    /**
+     * @param \DateTimeInterface|string|null  $date     A date or null to use the current time
+     * @param \DateTimeZone|string|false|null $timezone The target timezone, null to use the default, false to leave unchanged
+     */
+    public function formatDateTime(Environment $env, $date, string $locale = null, ?string $dateFormat = 'medium', ?string $timeFormat = 'medium', string $pattern = '', $timezone = null, string $calendar = 'gregorian'): string
```

Comment:
shouldn't be `$locale` the last argument?  (like in `localizeddate` filter)
most of the time, we configure the application to use a default locale (which is always the same on all templates) and vary the length of date per template. Repeating the first argument could be annoying:

ie. A sample of template I often meet
```diff
 <table class="list">
-  <tr><td>{{ item.createdAt|format_date(null, 'short') }}</td></tr>
-  <tr><td>{{ item.createdAt|format_date(dateFormat='short') }}</td></tr>
+  <tr><td>{{ item.createdAt|format_date('short') }}</td></tr>
 </table>
 <div class="detail">
-  <div>{{ item.createdAt|format_date(null, 'long') }}</div>
-  <div>{{ item.createdAt|format_date(dateFormat='long') }}</div>
+  <div>{{ item.createdAt|format_date('long') }}</div>
 </div>
```

Reply:
That makes sense, changed